java-grupa8/Dealership/src/main/java/ro/scii/java8/Customer.java

```diff
@@ -0,0 +1,395 @@
+        this.customerName = customerName;
+        this.budget = budget;
+    }
+
+    public int getBudget() {
+        return budget;
+    }
+
+
+    /**
+     * This method displays all the vehicles from a list
+     *
+     * @param list the list that is displayed
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void viewAllVehiclesList(List<ElectricVehicle> list) throws IOException, ClassNotFoundException {
+
+        for (ElectricVehicle v : list) {
+            System.out.println(v.getManufacturer() + "\n Model: " + v.getVehicleModel() + "\n Power: " + v.getMotorPower() + " kW " +
+                    "\t Battery: " + v.getBatteryCapacity() + " kWh" + "\t Range: " + v.getRange() + " km " + "\t Fast Charging: " + v.isHasFastCharging() +
+                    "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+        }
+        sortVehicles(list);
+    }
+
+
+    /**
+     * This method sorts a vehicle list based on parameters chosen by the customer from the keyboard
+     *
+     * @param list the list that is sorted
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void sortVehicles(List<ElectricVehicle> list) throws IOException, ClassNotFoundException {
+
+        PriceComparator priceComparator = new PriceComparator();
+        RangeComparator rangeComparator = new RangeComparator();
+        MotorPowerComparator motorPowerComparator = new MotorPowerComparator();
+
+        System.out.println("Do you want the vehicle list to be sorted by an attribute? y/n ");
+        String verifySort = s.next();
+        if (verifySort.equals("y")) {
+
+            System.out.println("Please input the sorting attribute: price = p, range = r, motor power = m");
+            String verifyAttribute = s.next();
+            switch (verifyAttribute) {
+
+                case "p":
+                    Collections.sort(list, priceComparator);
+                    break;
+
+                case "m":
+                    Collections.sort(list, motorPowerComparator);
+                    break;
+
+                case "r":
+                    Collections.sort(list, rangeComparator);
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Only \"p\", \"m\", or \"r\" options can be used.");
+
+            }
+            filterVehicles(list);
+
+        } else if (verifySort.equals("n")) {
+            filterVehicles(list);
+        } else {
+            throw new IllegalArgumentException("Only \"y\" or \"n\" options can be used.");
+        }
+    }
+
+    /**
+     * This method filters a vehicle list based on stock number and fast charging
+     *
+     * @param list the list that is filtered
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void filterVehicles(List<ElectricVehicle> list) throws IOException, ClassNotFoundException {
```

Comment:
This method does not belong to the customer. The customer might be calling it on a dealership.